native/dgesvd.go

```diff
@@ -0,0 +1,1241 @@
+				wrkbl = max(wrkbl, 3*m+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*m+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = m*m + wrkbl
+			} else if wantvs && wantuo {
+				// Path 5t
+				wrkbl = m + lwork_dgelqf
+				wrkbl = max(wrkbl, m+lwork_dorglq_m)
+				wrkbl = max(wrkbl, 3*m+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*m+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, 3*m+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = 2*m*m + wrkbl
+			} else if wantvs && wantuas {
+				// Path 6t
+				wrkbl = m + lwork_dgelqf
+				wrkbl = max(wrkbl, m+lwork_dorglq_m)
+				wrkbl = max(wrkbl, 3*m+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*m+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, 3*m+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = m*m + wrkbl
+			} else if wantva && wantun {
+				// Path 7t
+				wrkbl = m + lwork_dgelqf
+				wrkbl = max(wrkbl, m+lwork_dorglq_n)
+				wrkbl = max(wrkbl, 3*m+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*m+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = m*m + wrkbl
+			} else if wantva && wantuo {
+				// Path 8t
+				wrkbl = m + lwork_dgelqf
+				wrkbl = max(wrkbl, m+lwork_dorglq_n)
+				wrkbl = max(wrkbl, 3*m+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*m+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, 3*m+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = 2*m*m + wrkbl
+			} else if wantva && wantuas {
+				// Path 9t
+				wrkbl = m + lwork_dgelqf
+				wrkbl = max(wrkbl, m+lwork_dorglq_n)
+				wrkbl = max(wrkbl, 3*m+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*m+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, 3*m+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = m*m + wrkbl
+			}
+		} else {
+			// Path 10t, n > m
+			impl.Dgebrd(m, n, a, lda, s, dum, dum, dum, dum, -1)
+			lwork_dgebrd = int(dum[0])
+			maxwrk := 3*m + lwork_dgebrd
+			if wantvs || wantvo {
+				impl.Dorgbr(lapack.ApplyP, m, n, m, a, n, dum, dum, -1)
+				lwork_dorgbr_p = int(dum[0])
+				maxwrk = max(maxwrk, 3*m+lwork_dorgbr_p)
+			}
+			if wantva {
+				impl.Dorgbr(lapack.ApplyP, n, n, m, a, n, dum, dum, -1)
+				lwork_dorgbr_p = int(dum[0])
+				maxwrk = max(maxwrk, 3*m+lwork_dorgbr_p)
+			}
+			if !wantun {
+				maxwrk = max(maxwrk, 3*m+lwork_dorgbr_q)
+			}
+			maxwrk = max(maxwrk, bdspac)
+		}
+	}
+	maxwrk = max(maxwrk, minWork)
+	work[0] = float64(maxwrk)
+	if lwork == -1 {
+		return true
+	}
+
+	// Perform decomposition.
+	eps := dlamchE
+	smlnum := math.Sqrt(dlamchS) / eps
+	bignum := 1 / smlnum
+
+	// Scale A if max element outside range [smlnum, bignum]
+	anrm := impl.Dlange(lapack.MaxAbs, m, n, a, lda, dum)
+	iscl := 0
+	if anrm > 0 && anrm < smlnum {
+		iscl = 1
+		impl.Dlascl(lapack.General, 0, 0, anrm, smlnum, m, n, a, lda)
+	} else if anrm > bignum {
+		iscl = 1
+		impl.Dlascl(lapack.General, 0, 0, anrm, bignum, m, n, a, lda)
+	}
+
+	var ie int
+	if m >= n {
+		// If A has sufficiently more rows than columns, use the QR decomposition.
+		if m >= mnthr {
+			if wantun {
+				// Path 1
```

Comment:
Some of the fortran comment is useful here: "No left singular vectors to be computed."